### DISCUSSION TITLE
Comet Tails on Old Versions?

### DISCUSSION THREAD

**Discussion by creativspelerr** [Stellarium/stellarium#3650] [Q&A] [answered]
I'm doing a project in which being able to show comet tail on P1/Halley is a top priority.  I've read in some places (from the software author) that comet tails aren't available, but in other places it is documented that they are (with trouble shooting tips on how to get them to show back up).  From what I can piece together, they were first introduced in v0.13 and were removed sometime after v0.18.  I've downloaded those old versions and they don't show.  It could be that I need to activate some setting to make them visible, but my current guess is that those version were working with OpenGL/DirectX versions that are no longer compatible with today's graphics cards.
Any ideas how to get comet tails to render?

**Comment by gzotti** [comment]
Why should comet tails have been removed? Where have you seen that? If you have found some earlier developer stating around 2009 that comet tails are not available, this has changed, indeed almost 10 years ago in 0.13.0.
 
You can load the file 1000comets.ini to include about 1000 historical comets, first and foremost including 1P/Halley in apparitions since -1403. Of course, the absolute magnitudes of those early apparitions are crude estimates that may overshoot a bit. Tail tweaks are described in Appendix D.

**Reply by creativspelerr** [reply]
This seems to be from a developer saying that comet tails are not supposed to be included and people who are complaining about them not working are barking up the wrong tree.
https://badlyhonedbytes.wordpress.com/2011/09/26/stellarium-youre-using-it-wrong-comets-and-asteroids/

Like you said, I also found references to them being included in 0.13.0.
At any rate, if tails can be rendered in 23.x, all the better, I just need to figure out how!  I can search for 1P/Halley or Neowise and it finds them. I put in For Halley's Comet I put in 1910-8-5 T20:40:00 (Phoenix AZ) and it shows a magnitude of 2.13 in the info but no object.  
I tried going to Solar System Editor and did Import Elements from "Gideon van Buitenen" and selecting a few comets of interest but that didn't work.  
I found the file called ssystem_1000comets.ini  I haven't found any where to load that file or activate any comet related settings, so perhaps that's the step I'm missing.

**Reply by gzotti** [reply, marked as answer]
2011... In that text we can also read:

> All of this applies to Stellarium 0.11.0, released in July 2011. Some parts of the description may no longer apply to future versions.

Use a text search in the User Guide for "ssystem_1000comets.ini" or read the description of the Solar System Editor plugin. 
According to my orbital data for 1910 from the 1000comets file, on August 5th, 1910, 1P/Halley was mag10 and setting approximately with the sun. If you mean May 8th, it was a nice mag1 early-morning comet. 

If a comet is given as mag2, zoom in to see it. But if the info text says something about outdated elements, display of the object is suppressed to not evoke false expectations. Osculating orbital elements used for comets and other minor bodies are short-lived. You need at least 1 set of elements per perihelion passage for a comet.

**Reply by creativspelerr** [reply]
Got it!
The info in the upper left did say, "Elements for epoch 1986-01-15 probably outdated.  Consider updating data!"  Which probably contributed to August 5th reporting at magnitude 2.4, when yours reported mag 10.  
I had tried updating the information based on a procedure I found online for adding new comet data.  I was going a little deeper into the Solar System Editor, to the Solar System tab - Import orbital elements in MPC format.  Select bookmark - Gideon van Buitenen: comets.  And selecting individual comets.

What fixed it was going to the Solar System Editor, Configure, Configuration file tab, clicking on "import and add solar system minor objects from file" and selecting the 1000comets.ini.  

Now it shows the tails and the the May date and time data agrees with what you reported.

Edit: When I tried viewing the 1759 passage, I found the "probably outdated" message again, but I realized that in the search window it was pulling up my recent search history which had a single entry for "P1/Halley()" so I typed "halley" into the search window again now that I had imported the 1000comets.ini. This time it pulled up P/Halley (66), P/Halley (530), P/Halley (1759) and so on.  

For anyone in the future who has the same trouble I had, that's a good diagnostic.  The search window should list multiple entries, by year, and for it to display correctly, you have to match the viewing date and time to the instance in the search window.

Thanks so much!
